examples/platforms/cc2650/radio.c

```diff
@@ -0,0 +1,1696 @@
+ * - disables the radio core power domain
+ * - powers off the radio core power domain
+ * - switches the high frequency clock to the rcosc to save power
+ */
+static void rfCorePowerOff(void)
+{
+    rfCoreStopInt();
+
+    PRCMDomainDisable(PRCM_DOMAIN_RFCORE);
+    PRCMLoadSet();
+
+    while (!PRCMLoadGet());
+
+    PRCMPowerDomainOff(PRCM_DOMAIN_RFCORE);
+
+    while (PRCMPowerDomainStatus(PRCM_DOMAIN_RFCORE) != PRCM_DOMAIN_POWER_OFF);
+
+    if (OSCClockSourceGet(OSC_SRC_CLK_HF) != OSC_RCOSC_HF)
+    {
+        /* Request to switch to the RC osc for low power mode. */
+        OSCClockSourceSet(OSC_SRC_CLK_MF | OSC_SRC_CLK_HF, OSC_RCOSC_HF);
+        /* Switch the HF clock source (cc26xxware executes this from ROM) */
+        OSCHfSourceSwitch();
+    }
+}
+
+/**
+ * @brief sends the setup command string to the radio core
+ *
+ * Enables the clock line from the RTC to the RF core RAT. Enables the RAT
+ * timer and sets up the radio in IEEE mode.
+ *
+ * @return the value from the command status register
+ * @retval CMDSTA_Done the command was received
+ */
+static uint_fast16_t rfCoreSendEnableCmd(void)
+{
+    uint8_t ret;
+    bool interruptsWereDisabled;
+    /* turn on the clock line to the radio core */
+    HWREGBITW(AON_RTC_BASE + AON_RTC_O_CTL, AON_RTC_CTL_RTC_UPD_EN_BITN) = 1;
+
+    /* initialize the rat start command */
+    memset((void *)&sStartRatCmd, 0x00, sizeof(sStartRatCmd));
+    sStartRatCmd.commandNo = CMD_SYNC_START_RAT;
+    sStartRatCmd.condition.rule = COND_STOP_ON_FALSE;
+    sStartRatCmd.pNextOp = (rfc_radioOp_t *) &sRadioSetupCmd;
+    sStartRatCmd.rat0 = sRatOffset;
+
+    /* initialize radio setup command */
+    memset((void *)&sRadioSetupCmd, 0, sizeof(sRadioSetupCmd));
+    sRadioSetupCmd.commandNo = CMD_RADIO_SETUP;
+    sRadioSetupCmd.condition.rule = COND_NEVER;
+    /* initally set the radio tx power to the max */
+    sRadioSetupCmd.txPower = sCurrentOutputPower->value;
+    sRadioSetupCmd.pRegOverride = sIEEEOverrides;
+    sRadioSetupCmd.mode = 1;
+
+    interruptsWereDisabled = IntMasterDisable();
+
+    if ((ret = (RFCDoorbellSendTo((uint32_t)&sStartRatCmd) & 0xFF)) != CMDSTA_Done)
+    {
+        if (!interruptsWereDisabled)
+        {
+            IntMasterEnable();
+        }
+
+        return ret;
+    }
+
+    /* synchronously wait for the CM0 to stop executing */
+    while ((HWREG(RFC_DBELL_NONBUF_BASE + RFC_DBELL_O_RFCPEIFG) & IRQ_LAST_COMMAND_DONE) == 0x00);
+
+    if (!interruptsWereDisabled)
+    {
+        IntMasterEnable();
+    }
+
+    return sRadioSetupCmd.status;
+}
+
+/**
+ * @brief sends the shutdown command string to the radio core
+ *
+ * Powers down the frequency synthesizer and stops the RAT.
+ *
+ * @note synchronously waits until the command string completes.
+ *
+ * @return the status of the RAT stop command
+ * @retval DONE_OK the command string executed properly
+ */
+static uint_fast16_t rfCoreSendDisableCmd(void)
+{
+    uint8_t doorbell_ret;
```

Comment:
For the most part, OT uses camel case variables.  Snake case and camel case definitely shouldn't be mixed in the same function.